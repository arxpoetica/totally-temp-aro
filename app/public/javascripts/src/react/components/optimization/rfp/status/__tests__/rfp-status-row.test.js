/* global test expect */
import React from 'react'
import { shallow } from 'enzyme'
import RfpStatusRow from '../rfp-status-row.jsx'

const plan = {
  id: 42,
  name: 'RFP plan',
  createdBy: 4
}
const allPlanStates = [
  'UNDEFINED', 'START_STATE', 'INITIALIZED', 'STARTED', 'COMPLETED', 'CANCELED', 'FAILED'
]

const systemActors = {
  3: { id: 3, name: 'Public', description: 'Public users have read access to library items created during ETL', deleted: false, type: 'group' },
  4: { id: 4, firstName: 'Admin', lastName: 'User', type: 'user' }
}

// -----------------------------------------------------------------------------
test('When system actors are not loaded', () => {
  const component = shallow(
    <RfpStatusRow
      planId={plan.id}
      name={plan.name}
      createdById={plan.createdBy}
      status={'START_STATE'}
      reportDefinitions={[]}
      systemActors={{}}
      userId={42}
    />
  )
  expect(component).toMatchSnapshot()
})

// -----------------------------------------------------------------------------
test('When system actors are loaded', () => {
  const component = shallow(
    <RfpStatusRow
      planId={plan.id}
      name={plan.name}
      createdById={plan.createdBy}
      status={'START_STATE'}
      reportDefinitions={[]}
      systemActors={systemActors}
      userId={42}
    />
  )
  expect(component).toMatchSnapshot()
})

// -----------------------------------------------------------------------------
test('All plan states', () => {
  const component = shallow(
    <RfpStatusRow
      planId={plan.id}
      name={plan.name}
      createdById={plan.createdBy}
      status={'START_STATE'}
      reportDefinitions={[]}
      systemActors={systemActors}
      userId={42}
    />
  )
  allPlanStates.forEach(planState => {
    component.setProps({ status: planState })
    expect(component).toMatchSnapshot()
  })
})
