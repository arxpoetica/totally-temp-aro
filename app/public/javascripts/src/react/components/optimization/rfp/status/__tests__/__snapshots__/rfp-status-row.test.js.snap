// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All plan states 1`] = `
<tr
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <td>
    42
  </td>
  <td>
    RFP plan
  </td>
  <td>
    Admin User
  </td>
  <td>
    <div
      className="badge badge-danger"
    >
      UNDEFINED
    </div>
  </td>
  <td>
    <Connect(RfpReportDownloadCell)
      planId={42}
      reportDefinitions={Array []}
      userId={42}
    />
  </td>
</tr>
`;

exports[`All plan states 2`] = `
<tr
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <td>
    42
  </td>
  <td>
    RFP plan
  </td>
  <td>
    Admin User
  </td>
  <td>
    <div
      className="badge badge-dark"
    >
      START_STATE
    </div>
  </td>
  <td>
    <Connect(RfpReportDownloadCell)
      planId={42}
      reportDefinitions={Array []}
      userId={42}
    />
  </td>
</tr>
`;

exports[`All plan states 3`] = `
<tr
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <td>
    42
  </td>
  <td>
    RFP plan
  </td>
  <td>
    Admin User
  </td>
  <td>
    <div
      className="badge badge-dark"
    >
      INITIALIZED
    </div>
  </td>
  <td>
    <Connect(RfpReportDownloadCell)
      planId={42}
      reportDefinitions={Array []}
      userId={42}
    />
  </td>
</tr>
`;

exports[`All plan states 4`] = `
<tr
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <td>
    42
  </td>
  <td>
    RFP plan
  </td>
  <td>
    Admin User
  </td>
  <td>
    <div
      className="badge badge-primary"
    >
      STARTED
    </div>
  </td>
  <td>
    <Connect(RfpReportDownloadCell)
      planId={42}
      reportDefinitions={Array []}
      userId={42}
    />
  </td>
</tr>
`;

exports[`All plan states 5`] = `
<tr
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <td>
    42
  </td>
  <td>
    RFP plan
  </td>
  <td>
    Admin User
  </td>
  <td>
    <div
      className="badge badge-success"
    >
      COMPLETED
    </div>
  </td>
  <td>
    <Connect(RfpReportDownloadCell)
      planId={42}
      reportDefinitions={Array []}
      userId={42}
    />
  </td>
</tr>
`;

exports[`All plan states 6`] = `
<tr
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <td>
    42
  </td>
  <td>
    RFP plan
  </td>
  <td>
    Admin User
  </td>
  <td>
    <div
      className="badge badge-danger"
    >
      CANCELED
    </div>
  </td>
  <td>
    <Connect(RfpReportDownloadCell)
      planId={42}
      reportDefinitions={Array []}
      userId={42}
    />
  </td>
</tr>
`;

exports[`All plan states 7`] = `
<tr
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <td>
    42
  </td>
  <td>
    RFP plan
  </td>
  <td>
    Admin User
  </td>
  <td>
    <div
      className="badge badge-danger"
    >
      FAILED
    </div>
  </td>
  <td>
    <Connect(RfpReportDownloadCell)
      planId={42}
      reportDefinitions={Array []}
      userId={42}
    />
  </td>
</tr>
`;

exports[`When system actors are loaded 1`] = `
<tr
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <td>
    42
  </td>
  <td>
    RFP plan
  </td>
  <td>
    Admin User
  </td>
  <td>
    <div
      className="badge badge-dark"
    >
      START_STATE
    </div>
  </td>
  <td>
    <Connect(RfpReportDownloadCell)
      planId={42}
      reportDefinitions={Array []}
      userId={42}
    />
  </td>
</tr>
`;

exports[`When system actors are not loaded 1`] = `
<tr
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <td>
    42
  </td>
  <td>
    RFP plan
  </td>
  <td>
    UserID: 4
  </td>
  <td>
    <div
      className="badge badge-dark"
    >
      START_STATE
    </div>
  </td>
  <td>
    <Connect(RfpReportDownloadCell)
      planId={42}
      reportDefinitions={Array []}
      userId={42}
    />
  </td>
</tr>
`;
